docs/design/control-plane-resilience.md

```diff
@@ -0,0 +1,189 @@
+        1. reconstructible (e.g. using boot-from-snapshot,
+           boot-from-pre-configured-image or
+           boot-from-auto-initializing image).
+1. **High Availability:** This has the potential to increase
+   availability above the approximately "3 nines" level provided by
+   automated self-healing, but it's somewhat more complex, and
+   requires additional resources (e.g. redundant API servers and etcd
+   quorum members).  In environments where cloud-assisted automatic
+   self-healing might be infeasible (e.g. on-premise bare-metal
+   deployments), it also gives cluster administrators more time to
+   respond (e.g.  replace/repair failed machines) without incurring
+   system downtime.
+
+## Design and Status (as of December 2015)
+
+<table>
+<tr>
+<td>**Control Plane Component**</td>
+<td>**Resilience Plan**</td>
+<td>**Current Status**</td>
+</tr>
+<tr>
+<td>**API Server** </td>
+<td>
+
+Multiple stateless, self-hosted, self-healing API servers behind a HA
+load balancer, built out by the default "kube-up" automation on GCE,
+AWS and basic bare metal (BBM).  Detailed design of self-hosting and
+related bootstrapping and catastrophic failure recovery will be
+detailed in a separate design doc.
+
+</td>
+<td>
+
+No scripted self-healing or HA on GCE, AWS or basic bare metal
+currently exists in the OSS distro.  Self-healing and HA can be set up
+manually by following documented instructions, but this is not
+currently an automated process, and it is not tested as part of
+continuous integration.  So it's probably safest to assume that it
+doesn't actually work in practise.
+
+</td>
+</tr>
+<tr>
+<td>**Controller manager**</td>
+<td>
+
+Multiple self-hosted, self healing warm standby stateless controller
+managers with leader election and automatic failover of API server
+clients, built out by default "kube-up" automation.
+
+</td>
+<td>As above.</td>
+</tr>
+<tr>
+<td>**etcd**</td>
+<td>
+
+Multiple (3-5) etcd quorum members behind a load balancer with session
+affinity (to prevent clients from being bounced from one to another).
+Self-healing provided by remotely mounted persistent disks in
+non-performance critical, single-zone cloud deployments.  For
+performance critical installations, faster local SSD's should be used,
+in which case remounting on node failure is not an option, so etcd
+runtime configuration should be used to replace the failed machine.
+Similarly, for cross-zone self-healing, cloud persistent disks are
+zonal, so automatic runtime configuration is required.  Similarly,
+basic bare metal deployments cannot rely on remote persistent disks,
+so the same approach applies there.
+
+</td>
+<td>
+
+Somewhat vague instructions exist on how to set some of this up
+manually in a self-hosted configuration. But automatic bootstrapping
+and self-healing is not described (and is not implemented for the
+non-PD cases).  This all still needs to be automated and continuously
```

Comment:
I thought the API server section said API server is not self-healing (regardless of PD or non-PD).